sklearn/preprocessing/tests/test_data.py

```diff
@@ -2227,6 +2227,15 @@ def test_categorical_encoder_dtypes_pandas():
     assert_array_equal(enc.transform(X).toarray(), exp)
 
 
+def test_categorical_encoder_warning():
+    enc = CategoricalEncoder()
+    X = [['Male', 1], ['Female', 3]]
+    with warnings.catch_warnings(record=True) as w:
```

Comment:
Maybe use `sklearn.utils.testing.assert_no_warnings`

Reply:
`sklearn.utils.testing.assert_no_warnings` might be replaced by `numpy.testing.assert_no_warnings` since we dropped numpy 1.6, and it is based on `warnings.catch_warnings` for sure.

I just find it more readable to have:
```py
assert_no_warnings(enc.fit_transform, X)
```
than
```py
with warnings.catch_warnings(record=True) as w:
    enc.fit_transform(X)
assert len(w) == 0
```

Reply:
Right, makes sense, `sklearn.utils.testing.assert_no_warnings` it is then..

Reply:
Yes, I agree that using `numpy.testing.assert_no_warnings` will be more readable, updated.